core/common/src/Instant.kt

```diff
@@ -17,7 +17,11 @@ import kotlin.time.*
  * in different time zones: someone in Tokyo would think its already `1970-01-01` several hours earlier than someone in
  * Berlin would. To represent such entities, use [LocalDateTime].
  * In contrast, "the moment the clocks in London first showed 00:00 on Jan 1, 2000" is a specific moment
- * in time, as is "1970-01-01, 00:00:00 UTC", and so can be represented as an [Instant].
+ * in time, as is "1970-01-01, 00:00:00 UTC+0", and so it can be represented as an [Instant].
+ *
+ * `Instant` uses UTC-SLS (smeared leap second) time scale. This time scale doesn't contain instants
+ * corresponding to leap seconds, but instead "smears" positive and negative leap seconds among the last 1000 seconds
+ * of the day when a leap second happens.
```

Comment:
This whole section, I think, is out-of-place here, as the implementation of `Instant` does not reflect the SLS aspect.

Additionally, it should be qualified with "on the JVM, if the implementation permits it", or, better yet, simply removed, both from here and from the README.
On the JS, just `Date` is used: https://github.com/js-joda/js-joda/blob/0702890538ab460a446aba89622878fb70fc94e2/packages/core/src/Clock.js#L218
Searching for the JVM implementation of UTC-SLS, I only find mentions of it being specified, but not supported:  https://stackoverflow.com/a/35465689 https://stackoverflow.com/a/31162603
On the Native, well, there's nothing in the vein of UTC-SLS either.

Reply:
It's the time scale we have chosen for Instants. The section answers the question "where are my leap seconds". The fact that the platform clocks don't provide this behavior can be explained by their inaccuracy.

Reply:
> length of the ISO-8601 calendar day (3.1.2.11) depends on the leap seconds, so we can't even claim that `DateTimePeriod` corresponds to some ISO-8601 entity or a combination of them.

Now that I think of it, it seems fine. So, only the `Instant` parsing/formatting is problematic.

Reply:
I don't think ISO mandates the use of UTC scale with leap seconds. As usual this is handled by an agreement between the interchange parties.

Reply:
Seems fairly conclusive to me.

3.1.1.13. **UTC of day** is the time of day *in UTC*.
3.2.5. "Z" indicates that a time of day is represented as *UTC of day*.

Reply:
Again, I believe it is possible to use ISO format to interchange UTC-SLS instants without converting them first to UTC. It just has to be agreed upon, and the appropriate place to state that agreement would indeed be where we parse and format values from and to ISO string representations.

Reply:
Could you point to a particular section where this is permitted? I traced `"Z"` meaning exactly UTC to the base definitions, but maybe there are some rules overriding this that I missed.

Reply:
No, that's my interpretation of it.

Reply:
Sure, but what part of the standard is open to interpretation?

Reply:
Historical note: after an internal discussion, we decided that this is fine, being the de-facto way the ISO-8601 strings are typically used.